sources/models/src/modeled_types/ecs.rs

```diff
@@ -291,3 +291,69 @@ mod test_ecs_agent_image_pull_behavior {
         }
     }
 }
+
+/// ECSDurationValue represents a string that contains a valid ECS duration value
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ECSDurationValue {
+    inner: String,
+}
+
+lazy_static! {
+    pub(crate) static ref ECS_DURATION_VALUE: Regex =
+        Regex::new(r"^(([0-9]+\.)?[0-9]+h)?(([0-9]+\.)?[0-9]+m)?(([0-9]+\.)?[0-9]+s)?(([0-9]+\.)?[0-9]+ms)?(([0-9]+\.)?[0-9]+(u|µ){1}s)?(([0-9]+\.)?[0-9]+ns)?$").unwrap();
```

Comment:
nit: regex101 called the `{1}` repetition "useless", and suggested using a non-capturing group for `(u|µ)` instead:
```suggestion
        Regex::new(r"^(([0-9]+\.)?[0-9]+h)?(([0-9]+\.)?[0-9]+m)?(([0-9]+\.)?[0-9]+s)?(([0-9]+\.)?[0-9]+ms)?(([0-9]+\.)?[0-9]+(?:u|µ)s)?(([0-9]+\.)?[0-9]+ns)?$").unwrap();
```